sympy/utilities/lambdify.py

```diff
@@ -424,6 +454,13 @@ def lambdify(args, expr, modules=None, printer=None, use_imps=True,
     # Create lambda function.
     lstr = lambdastr(args, expr, printer=printer, dummify=dummify)
     flat = '__flatten_args__'
+    imp_mod_lines = []
+    for mod, keys in (getattr(printer, 'module_imports', None) or {}).items():
```

Comment:
That would be a problem in the printer already then (unlikely and would require two different _print methods to output functions with same name from different modules). The order in the ``modules`` kwarg to lambdify does still matter though.